packages/keyring/src/pair/index.ts

```diff
@@ -44,7 +44,13 @@ const TYPE_SIGNATURE = {
 const TYPE_ADDRESS = {
   ecdsa: (p: Uint8Array) => p.length > 32 ? blake2AsU8a(p) : p,
   ed25519: (p: Uint8Array) => p,
-  ethereum: (p: Uint8Array) => keccakAsU8a(secp256k1Expand(p)),
+  ethereum: (p: Uint8Array) => {
+    if (p.length === 20) {
+      return p;
+    } else {
+      return hexToU8a(ethereumEncode(keccakAsU8a(secp256k1Expand(p))));
+    }
+  },
```

Comment:
```suggestion
  ethereum: (p: Uint8Array) => p.length === 20 ? p : keccakAsU8a(secp256k1Expand(p)),
```

This seems like the only code change required (before the addition of the specific test as per the first commit)

Reply:
See https://github.com/polkadot-js/common/compare/jg-eth-address?expand=1 for the one line change + tests (lifted from here, only addition is the removal of the toLowerCase on the address)